src/environment.cpp

```diff
@@ -85,6 +87,181 @@ float Environment::getTimeOfDayF()
 	return m_time_of_day_f;
 }
 
+/*
+	Check if a node is pointable
+*/
+inline bool isPointableNode(const MapNode &n,
+			    INodeDefManager *nodedef , bool liquids_pointable)
+{
+	const ContentFeatures &features = nodedef->get(n);
+	return features.pointable ||
+	       (liquids_pointable && features.isLiquid());
+}
+
+PointedThing Environment::continueRaycast(RaycastState &state)
```

Comment:
Also pointedThing should be also returned by reference using the function member to prevent a memory copy